### ISSUE TITLE
[vim-mode] ctrl-b does not keep the cursor within the window, although ctrl-f does

### ISSUE THREAD

**Issue by graelo** [zed-industries/zed#17687]
### Check for existing issues

- [X] Completed

### Describe the feature

When scrolling forward, zed implements perfectly vim's behavior of keeping the cursor within the visible part of the buffer (aka the window).

However when scrolling backward, the cursor remains at its original position, outside the window. This makes it impossible (at least awkward) to perform visual selections over regions larger than a window when scrolling back.

Can you fix this please?

Thanks!

PS: maybe I should have filed this as a bug, but it sounded a bit too strong.

**Comment by maan2003** [comment]
I don't see that behavior in down scroll with trackpad

**Comment by uminoros** [comment]
I am experiencing the same issue while using Zed 0.152.3 on macOS.

![output](https://github.com/user-attachments/assets/8c213dd5-3612-4086-bca4-5d6e60cf23a5)

In the GIF, I first used ``ctrl+f`` to move the cursor to the bottom of the file, and it worked correctly.
Next, I tried to use ``ctrl+b`` to move the cursor to the top of the file, but only the window moved to the top, while the cursor remained at the bottom.

I remember ``ctrl+b`` working correctly in the previous version.
